### PR TITLE
OSASINFRA-3551: Updating the NovaComputeCpu Dedicated and Shared Set in nfv config

### PR DESCRIPTION
**Pull request by MiguelCarpio**
NFV jobs are failing due to insufficient NovaComputeCpuShared cores in the NFV Cloud to deploy controlplane and bootstrap nodes. Changes applied:

- Increasing NovaComputeCpuSharedSet to 60 cores. So isolated_cores (OvsPmdCoreList  + NovaComputeCpuDedicatedSet) will reduced from 56 cores to 36 (allowing at least 4 NFV compute nodes or 4 NFV jobs). 
- Removing the assigned OvsPmdCoreList from the NovaComputeCpuDedicatedSet list. NovaComputeCpuDedicatedSet: Exclude all cores from the OvsPmdCoreList. Pair the sibling threads together. Include all remaining cores.

### CODE REVIEW

File: configs/nfv.yaml [hunk 1]
```diff
@@ -28,15 +28,15 @@ sriov_nova_pci_passthrough:
     physical_network: "mellanox-sriov"
 sriov_nic_numvfs: 8
 dpdk_interface: enp193s0f0
-kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=420 iommu=pt amd_iommu=on isolcpus=10-23,58-71,34-47,82-95"
-tuned_isolated_cores: 10-23,58-71,34-47,82-95
+kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=420 iommu=pt amd_iommu=on isolcpus=15-23,63-71,39-47,87-95"
```
[the remark this answers is not in the document]
Comment (EmilienM) [reply configs/nfv.yaml:31]: I found my answer here I think:

https://docs.redhat.com/en/documentation/red_hat_openstack_platform/17.1/html-single/configuring_network_functions_virtualization/index#example-ceph-config-file_hci-nfvsub

```
To calculate hugepages, subtract the value of the NovaReservedHostMemory parameter from total memory. 
```

and `NovaReservedHostMemory`:
```
Reserves memory in MB for tasks on the host. NovaReservedHostMemory is the reserved_host_memory_mb value for the Compute node in nova.conf
``` 

So I think that number could be increased a bit to give more hugepages? Or `NovaReservedHostMemory` could be increased to give more RAM to the system.

[the remark this answers is not in the document]
Comment (stephenfin) [reply configs/nfv.yaml:31]: How much memory is specified in the flavour for the NFV worker? From what @MiguelCarpio is saying about being able to have 4 machines, I'm guessing there's 8 vCPUs. To work out the amount of hugepages needed, first figure out the amount of hugepages needed by an instance (`flavor.memory_mb / 1024`), then multiply that by the amount of instances you expect to host, and finally add the amount of pages needed for OVS-DPDK. For example, if the flavour has `memory_mb = 16384` and you want to schedule up to 4 instances at any given time:

```
16384 / 1024 = 16  # flavour.memory_mb / 1024
16 * 4 = 64        # no. hugepages per instance * no. instances
64 + 2 + 2 = 68    # no. hugepages for all instances + no. hugepages for OVS-DPDK
```

If that 16GB figure is correct, then I think we could actually drop the amount of hugepages assigned here to free up memory for the host and master machines (which presumably don't use hugepages)?

File: configs/nfv.yaml [hunk 2]
```diff
@@ -28,15 +28,15 @@ sriov_nova_pci_passthrough:
...
+tuned_isolated_cores: 15-23,63-71,39-47,87-95
 extra_heat_params:
   CinderRbdFlattenVolumeFromSnapshot: true
-  NovaComputeCpuDedicatedSet: "10-23,58-71,34-47,82-95"
+  NovaComputeCpuDedicatedSet: "16-23,63-71,40-47,87-95"
   NovaReservedHostMemory: 4096
-  NovaComputeCpuSharedSet: "0-9,48-57,24-33,72-81"
+  NovaComputeCpuSharedSet: "0-14,48-62,24-38,72-86"
```
Comment (stephenfin) [inline configs/nfv.yaml:37]: This is a nit, but this gives us 34 pCPUs. We only need 32 (assuming our flavor uses 8 vCPUs). How about:

```suggestion
  NovaComputeCpuDedicatedSet: "16-23,64-71,40-47,88-95"
  NovaReservedHostMemory: 4096
  NovaComputeCpuSharedSet: "0-14,48-63,24-38,72-87"
```
```

File: configs/nfv.yaml [hunk 3]
```diff
@@ -28,15 +28,15 @@ sriov_nova_pci_passthrough:
...
   OvsDpdkSocketMemory: "2048,2048"
-  OvsPmdCoreList: "10,34"
+  OvsPmdCoreList: "15,38"
```
Comment (stephenfin) [inline configs/nfv.yaml:39]: I think you meant:

```suggestion
  OvsPmdCoreList: "15,39"
```

Comment (MiguelCarpio) [reply configs/nfv.yaml:39]: Yes! 
NUMA 0 -> CORE 20 -> CPU 15
NUMA 1 -> CORE 20 -> CPU 39

File: configs/nfv.yaml (hunk 2 above)
Comment (MiguelCarpio) [reply configs/nfv.yaml:37]: Ok!

File: configs/nfv.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (MiguelCarpio) [reply configs/nfv.yaml:31]: Ok, so move from 420 to 68

**Comment by MiguelCarpio** [comment]
Hey @EmilienM @stephenfin I've updated the PR with your recommendations. How about it now?

File: configs/nfv.yaml [hunk 4]
```diff
@@ -28,15 +28,15 @@ sriov_nova_pci_passthrough:
     physical_network: "mellanox-sriov"
 sriov_nic_numvfs: 8
 dpdk_interface: enp193s0f0
-kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=420 iommu=pt amd_iommu=on isolcpus=10-23,58-71,34-47,82-95"
-tuned_isolated_cores: 10-23,58-71,34-47,82-95
+kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=68 iommu=pt amd_iommu=on isolcpus=15-23,64-71,39-47,88-95"
```
Comment (stephenfin) [inline configs/nfv.yaml:31]:
```suggestion
# We split cores between different pools like so:
#
#  0-14: vCPUs
#    15: OVS-DPDK
# 16-23: pCPUs
# 24-38: vCPUs
#    39: OVS-DPDK
# 40-47: pCPUs
# 48-63: vCPUs
# 64-71: pCPUs
# 72-87: vCPUs
# 88-95: pCPUs
kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=68 iommu=pt amd_iommu=on isolcpus=15-23,64-71,39-47,88-95"
```

**Review by stephenfin — COMMENTED** [review]
/lgtm

I have one comment inline but it's a nit.

**Pull request merged**